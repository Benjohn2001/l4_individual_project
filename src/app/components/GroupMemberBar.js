import { Component } from "react";
import PropTypes from 'prop-types';
import { TouchableOpacity, Text, Image, View, Vibration } from "react-native";
import { Feather } from '@expo/vector-icons';

class GroupMemberBar extends Component {
    render(){
        const { title, onPress, avatar, color} = this.props;
		return (
            <View className="items-center py-1">
                <TouchableOpacity onPress={onPress} className="bg-secondaryPurple h-14 w-80 rounded-2xl mx-1.5">
                    <View className="flex-row justify-between items-center mx-3 pt-1">
                        <Image 
                            className="w-12 h-12 rounded-full" 
                            source = {avatar}
                        />
                        <Text  className="text-black text-lg text-center pt-1">{title}</Text>
                        <View className="h-8 w-8 rounded-full" style={{backgroundColor: color}}/>
                    </View>
                </TouchableOpacity>
            </View>
        );
    }
}

GroupMemberBar.propTypes = {
    title: PropTypes.string.isRequired,
    onPress: PropTypes.func.isRequired,
    avatar: PropTypes.number.isRequired,
    color: PropTypes.string.isRequired
};

export default GroupMemberBar;