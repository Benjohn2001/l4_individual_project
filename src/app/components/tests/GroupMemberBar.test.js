import { render, screen, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import GroupMemberBar from "../GroupMemberBar";

test("Check GroupMemberBar renders correctly and onpress works", () => {
  const onPressMock = jest.fn();
  render(
    <GroupMemberBar
      title="press"
      onPress={() => {
        onPressMock();
      }}
      color="red"
    />
  );
  fireEvent.press(screen.getByText("press"));
  expect(onPressMock);
  const domTree = renderer
    .create(
      <GroupMemberBar
        title="press"
        onPress={() => {
          onPressMock();
        }}
        color="red"
      />
    )
    .toJSON();
  expect(domTree).toMatchSnapshot();
});
