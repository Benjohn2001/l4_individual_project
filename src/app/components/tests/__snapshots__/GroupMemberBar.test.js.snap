// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check GroupMemberBar renders correctly and onpress works 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "paddingBottom": 4,
        "paddingTop": 4,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#E7E7FF",
        "borderBottomLeftRadius": 16,
        "borderBottomRightRadius": 16,
        "borderTopLeftRadius": 16,
        "borderTopRightRadius": 16,
        "height": 56,
        "marginLeft": 6,
        "marginRight": 6,
        "opacity": 1,
        "width": 320,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "justifyContent": "space-between",
          },
          Object {
            "alignItems": "center",
          },
          Object {
            "marginLeft": 12,
            "marginRight": 12,
          },
          Object {
            "paddingTop": 4,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "uri",
          }
        }
        style={
          Array [
            Object {
              "width": 48,
            },
            Object {
              "height": 48,
            },
            Object {
              "borderBottomLeftRadius": 9999,
              "borderBottomRightRadius": 9999,
              "borderTopLeftRadius": 9999,
              "borderTopRightRadius": 9999,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Object {
              "fontSize": 18,
              "lineHeight": 28,
            },
            Object {
              "textAlign": "center",
            },
            Object {
              "paddingTop": 4,
            },
          ]
        }
      >
        press
      </Text>
      <View
        style={
          Array [
            Object {
              "height": 32,
            },
            Object {
              "width": 32,
            },
            Object {
              "borderBottomLeftRadius": 9999,
              "borderBottomRightRadius": 9999,
              "borderTopLeftRadius": 9999,
              "borderTopRightRadius": 9999,
            },
            Object {
              "backgroundColor": "red",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
